### PR TITLE
libusbp: init at 1.3.0 AND pololu-tic: init at 1.8.1

### PR DESCRIPTION
**Pull request by bzizou**
###### Description of changes

The pololu libusbp is a dependency for the pololu tic stepper motor controler driver that will come in another pull request
https://www.pololu.com/category/212/tic-stepper-motor-controllers

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libusbp/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, udev
+, cmake
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
```
Comment (doronbehar) [inline pkgs/development/libraries/libusbp/default.nix:9]: I'd recommend `stdenv.mkDerivation(finalAttrs: {`.

**Review by doronbehar — CHANGES_REQUESTED** [review]
I'd recommend also using `pkgs/by-name`, and add the stepper motor package you mentioned in this PR as well, in a separate commit.

File: pkgs/development/libraries/libusbp/default.nix (hunk 1 above)
Comment (bzizou) [reply pkgs/development/libraries/libusbp/default.nix]: All done:
- using pkgs/by-name
- added the stepper motor package "pololu-tic" as a separate commit
- using finalAttrs

File: pkgs/by-name/li/libusbp/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}:
+
+stdenv.mkDerivation(finalAttrs: {
+  pname = "libusbp";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "pololu";
+    repo = "libusbp";
+    rev = "refs/tags/${finalAttrs.version}";
```
Comment (doronbehar) [inline pkgs/by-name/li/libusbp/package.nix:16]: Why not simply:

```suggestion
    rev = finalAttrs.version;
```

File: pkgs/by-name/li/libusbp/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "sha256-60xpJ97GlqEcy2+pxGNGPfWDnbIFGoPXJijaErOBXQs=";
```
Comment (doronbehar) [inline pkgs/by-name/li/libusbp/package.nix:17]: Nowadays this is preferred:

```suggestion
    hash = "sha256-60xpJ97GlqEcy2+pxGNGPfWDnbIFGoPXJijaErOBXQs=";
```

File: pkgs/by-name/li/libusbp/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [ cmake pkg-config ];
```
Comment (doronbehar) [inline pkgs/by-name/li/libusbp/package.nix:22]: Personally, I prefer this formatting:

```suggestion
  nativeBuildInputs = [
    cmake
    pkg-config
  ];
```

As it makes future diffs smaller and easier to review.

File: pkgs/by-name/li/libusbp/package.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  propagatedBuildInputs = [ udev ];
```
Comment (doronbehar) [inline pkgs/by-name/li/libusbp/package.nix:23]: Same here:

```suggestion
  propagatedBuildInputs = [
    udev
  ];
```

File: pkgs/by-name/po/pololu-tic/package.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    repo = "pololu-tic-software";
+    rev = "refs/tags/${finalAttrs.version}";
+    sha256 = "sha256-C/v5oaC5zZwm+j9CbFaDW+ebzHxPVb8kZLg9c0HyPbc=";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [ cmake pkg-config qt5.wrapQtAppsHook ];
+  propagatedBuildInputs = [ libusbp ];
+  buildInputs = [ qt5.qtbase ];
```
Comment (doronbehar) [inline pkgs/by-name/po/pololu-tic/package.nix:25]: Same comments :)

**Review by doronbehar — COMMENTED** [review]
Small styling comments, should be good to go after these.

**Review by doronbehar — APPROVED** [review]

**Pull request merged**